### PR TITLE
Feat/auto estimate sign deploy account

### LINKED ISSUE #443: Add `auto_estimate` parameter to `AccountClient.sign_deploy_account_transaction`

### PR DESCRIPTION
**Pull request by war-in**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Closes #443

* **What is the current behavior?** (You can also link to an open issue here)

### CODE REVIEW

File: starknet_py/net/account/account_client.py [hunk 1]
```diff
@@ -318,9 +318,12 @@ async def _get_max_fee(
...
             )
 
-        if isinstance(transaction, Declare) and transaction.version != 1:
+        if (
+            isinstance(transaction, (Declare, DeployAccount))
+            and transaction.version != 1
+        ):
             raise ValueError(
-                "Estimating fee for Declare transactions with versions other than 1 is not supported."
+                "Estimating fee for Declare/DeployAccount transactions with versions other than 1 is not supported."
```
Comment (cptartur) [inline starknet_py/net/account/account_client.py:326]: I wonder if we should be raising more specific errors per transaction type? But perhaps it's not needed.

**Review by cptartur — APPROVED** [review]

File: starknet_py/net/account/account_client.py (hunk 1 above)
Comment (war-in) [reply starknet_py/net/account/account_client.py:326]: I think it is not needed. It is the same error

Comment (war-in) [reply starknet_py/net/account/account_client.py:326]: I can do it this way:
`f"Estimating fee for {type(transaction)} transactions..."`

Comment (cptartur) [reply starknet_py/net/account/account_client.py:326]: Okay, let's merge it then.

**Pull request merged**